File: ShadowFrameworkLite20/src/shadow/pipeline/expression/SFExpressionGeneratorKeeper.java
package shadow.pipeline.expression;

public class SFExpressionGeneratorKeeper {

	private static SFExpressionGeneratorKeeper keeper=new SFExpressionGeneratorKeeper();
	
	private SFExpressionGeneratorKeeper(){
		
	}

	public static SFExpressionGeneratorKeeper getKeeper() {
		return keeper;
	}

	private SFIExpressionGenerator generator;

	public SFIExpressionGenerator getGenerator() {
		return generator;
	}

	public void setGenerator(SFIExpressionGenerator generator) {
		this.generator = generator;
	}
	
}

File: trunk/ShadowFrameworkJS/src/shadow/pipeline/expression/SFExpressionGeneratorKeeper.js
//Java to JS on 21/03/2012

function SFExpressionGeneratorKeeper(){
	this.generator=new SFBasicExpressionGenerator();
}

SFExpressionGeneratorKeeper.prototype["getGenerator"]=function(){
			return this.generator;
		};

SFExpressionGeneratorKeeper.prototype["setGenerator"]=function(generator){
			this.generator = generator;
		};

//Was originally a Singleton
var SFExpressionGeneratorKeeper_keeper=new SFExpressionGeneratorKeeper();

function SFExpressionGeneratorKeeper_getKeeper(){
	return SFExpressionGeneratorKeeper_keeper;
}
